packages/lib/survey/service.ts

```diff
@@ -849,6 +849,196 @@ export const createSurvey = async (
   }
 };
 
+export const migrateSurveyToOtherEnvironment = async (
+  surveyId: string,
+  environmentId: string,
+  targetEnvironmentId: string,
+  userId: string
+): Promise<Prisma.Survey> => {
+  validateInputs([environmentId, ZId], [surveyId, ZId], [targetEnvironmentId, ZId], [userId, ZId]);
+
+  // Start a transaction
+  const updatedSurvey = await prisma.$transaction(async (prisma) => {
+    try {
+      // Fetch required resources
+      const existingSurvey = await getSurvey(surveyId);
+      if (!existingSurvey) throw new ResourceNotFoundError("Survey", surveyId);
+
+      const [targetEnvironment, targetProduct] = await Promise.all([
+        getEnvironment(targetEnvironmentId),
+        getProductByEnvironmentId(targetEnvironmentId),
+      ]);
+
+      if (!targetEnvironment) throw new ResourceNotFoundError("Environment", targetEnvironmentId);
+      if (!targetProduct) throw new ResourceNotFoundError("Product", targetEnvironmentId);
+
+      // Prepare survey update data (excluding triggers)
+      const surveyUpdateData: Prisma.SurveyUpdateInput = {
+        environment: {
+          connect: { id: targetEnvironmentId },
+        },
+        languages: {
+          create: existingSurvey.languages.map((surveyLanguage) => ({
+            language: {
+              connectOrCreate: {
+                where: {
+                  productId_code: { code: surveyLanguage.language.code, productId: targetProduct.id },
+                },
+                create: {
+                  code: surveyLanguage.language.code,
+                  alias: surveyLanguage.language.alias,
+                  productId: targetProduct.id,
+                },
+              },
+            },
+            default: surveyLanguage.default,
+            enabled: surveyLanguage.enabled,
+          })),
+        },
+        segment: existingSurvey.segment
+          ? existingSurvey.segment.isPrivate
+            ? {
+                create: {
+                  title: existingSurvey.segment.title,
+                  isPrivate: true,
+                  filters: existingSurvey.segment.filters,
+                  environment: { connect: { id: targetEnvironmentId } },
+                },
+              }
+            : {
+                connectOrCreate: {
+                  where: {
+                    title_environmentId: {
+                      title: existingSurvey.segment.title,
+                      environmentId: targetEnvironmentId,
+                    },
+                  },
+                  create: {
+                    title: existingSurvey.segment.title,
+                    isPrivate: false,
+                    filters: existingSurvey.segment.filters,
+                    environment: { connect: { id: targetEnvironmentId } },
+                  },
+                },
+              }
+          : undefined,
```

Comment:
This logic has an issue: when the target environment has a segment with the same name as the segment used in the survey before migration, it will try to link the already existing segment to the migrated survey, even when they have different filters. 
Please check the implementation of the segment, trigger and language migration from the `copySurveyToOtherEnvironment` service.